TACC/Core-Portal
# task/FP-1902: Common Searchbar component

## Overview

Created a common searchbar component similar to the [DataFilesSearchbar](https://github.com/TACC/Core-Portal/tree/main/client/src/components/DataFiles/DataFilesSearchbar) component. The idea was to take existing functionality of the DataFilesSearchbar component and make it extendable so that other components can utilize the searchbar with their own parameters. This new component can replace existing uses of the DataFilesSearchbar component and can be used in new pages as well (Job search for example)


## Related

* [FP-1902](https://jira.tacc.utexas.edu/browse/FP-1902?filter=-1)

## Changes

- Added a new Searchbar compoment in _common including css module and test file. 
- Changed the DataFilesListing component to use the new common Searchbar component

Changes from the DataFilesSearchbar implementation include:
- Added **dataType** prop. This is used to determine what type of data will be searched for using this searchbar (e.g Files, Jobs etc.)
- Added **filterTypes** prop. Previously in the DataFilesSearchbar component the filters that would be populated in the dropdown were defined in the same file. In order to make this functionality more generic I added this filterTypes prop. Wherever the searchbar is used that component can pass in an array of strings that would be populated in the filter dropdown. If nothing is passed into this prop then the filter dropdown will not display. 
- Added **infinitScroll** prop. This is a boolean that is used to identify if a page uses infinite scrolling or pagination to display additional data. The SiteSearch page for example uses pagination while the DataFiles page uses infinite scrolling. Previously, a check was made to the siteSearch boolean in order to determine this. This prop makes this functionality more generic and can be used by other components as well 
- Changed **File Types** label to **Filter** so that it applies to different types of filters

## Testing

1. Open the portal and go to the Data Files section. 
2. Ensure search functionality works the same as before

## UI

DataFilesSearchbar component: 

![Screen Shot 2022-10-24 at 2 14 58 PM](https://user-images.githubusercontent.com/23081932/197607386-3e996a29-7a75-4e0f-8344-f6a0521f9f24.png)

New Searchbar component:

![Screen Shot 2022-10-24 at 2 14 28 PM](https://user-images.githubusercontent.com/23081932/197607425-d5447d3e-b93f-410f-bb1e-eb9c0198e738.png)


## Notes
Once tested and confirmed functional will likely have to clean up and remove the components it replaces. That would include the [DataFilesSearchbar](https://github.com/TACC/Core-Portal/tree/FP-1902-common-searchbar-component/client/src/components/DataFiles/DataFilesSearchbar) and [DataFilesProjectsSearchbar](https://github.com/TACC/Core-Portal/tree/FP-1902-common-searchbar-component/client/src/components/DataFiles/DataFilesProjectsList/DataFilesProjectsSearchbar) components
